### PR TITLE
Fix stock available after add product in order

### PR DESCRIPTION
**Pull request by MounirBoukhris**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | Fix the product's available quantity, also call "updateQuantity" function regardless the value of "Available quantities" option (in product->edit->quantities tab)
| Type?         | bug fix 
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/PSCSX-9230
| How to test?  | View an order in BO, add a new product in this order, set a quantity  for this product, and check if the available quantity of product is correct or not after adding this product.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8233)

### CODE REVIEW

**Comment by hibatallahAouadni** [comment]
@marionf could you please test it?
Thanks in advance :)

File: classes/order/OrderDetail.php [hunk 1]
```diff
@@ -468,10 +468,7 @@ protected function setVirtualProductInformation($product)
     protected function checkProductStock($product, $id_order_state)
     {
         if ($id_order_state != Configuration::get('PS_OS_CANCELED') && $id_order_state != Configuration::get('PS_OS_ERROR')) {
-            $update_quantity = true;
-            if (!StockAvailable::dependsOnStock($product['id_product'])) {
-                $update_quantity = StockAvailable::updateQuantity($product['id_product'], $product['id_product_attribute'], -(int)$product['cart_quantity']);
-            }
+            $update_quantity = StockAvailable::updateQuantity($product['id_product'], $product['id_product_attribute'], -(int)$product['cart_quantity']);
```
Comment (hibatallahAouadni) [inline classes/order/OrderDetail.php:471]: @Quetzacoalt91 code review please. 
Thanks in advance :)

**Comment by marionf** [comment]
Hello @hibatallahAouadni 
I will test after the code review, this avoid wasting time if changes are requested ;)

**Comment by hibatallahAouadni** [comment]
@eternoendless this one is supposed to be merged in 1.6.1.17 but till now it is not considered to be included in patch 1.6.1.18, why is that?

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Thank you @MounirBoukhris

**Pull request merged**